### PR TITLE
Feature/merge exclude columns

### PR DESCRIPTION
**Pull request by dave-connors-3**
resolves #5260 (dbt-core issue)

### Description

This PR adds integration tests for the associated logic change in dbt-core [PR #5457](https://github.com/dbt-labs/dbt-core/pull/5457)

### Related PRs:
- https://github.com/dbt-labs/dbt-core/pull/6700
- https://github.com/dbt-labs/dbt-bigquery/pull/227
- https://github.com/dbt-labs/dbt-snowflake/pull/206
- https://github.com/dbt-labs/dbt-spark/pull/601

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-bigquery next" section.

### CODE REVIEW

**Comment by VersusFacit** [comment]
I assume this will be unpinned and sent back to main before merge?

**Comment by dave-connors-3** [comment]
@VersusFacit that's the idea! once dbt-core is rocking and rolling, we can unpin and re-test

**Review by colin-rogers-dbt — APPROVED** [review]

File: tests/integration/simple_copy_test/test_simple_copy.py [hunk 1]
```diff
@@ -93,5 +93,4 @@ def test__bigquery__incremental_merge_columns(self):
             "seed-paths": ["seeds-merge-cols-update"]
         })
         self.seed_and_run()
-        self.assertTablesEqual("incremental_update_cols", "expected_result")
-
+        self.assertTablesEqual("incremental_update_cols", "expected_result")
```
Comment (VersusFacit) [inline tests/integration/simple_copy_test/test_simple_copy.py:96]: good auto formatting lol

**Review by VersusFacit — APPROVED** [review]
LGTM (barring the changeback on the dev requirements once we complete hashing out the adapter zone class)

**Pull request merged**